### PR TITLE
feat: simplify asset inheritance

### LINKED ISSUE #126: Schema inheritance optimizations
As an extension of https://github.com/carbon-design-system/carbon-platform/issues/7, there are a few things that we should consider when putting together the final version of our library and asset schema. Currently, it's structured like this:

```yml
id: accordion
inherits:
  asset: 'carbon-styles@latest/accordion'
  properties:
    - name
    - description
    - thumbnailPath
    - type
    - platform
status: stable
framework: react
```

## Auto detection

Can we crawl the package.json's to determine when inheritance is happening so it doesn't have to be explicitly stated? As package.json dependencies are just on the "library" level, this might not be possible because then how do we map assets within each library? It may be too restrictive for asset ids to match, or if asset names need to match if we slugify those. Also, there's no deterministic way to match assets by path as directory and file names may be different per language conventions.

On a similar note, can we auto-detect version based on package.json? That seems much more attainable.

## "Inheritable" properties

Do we want to specify which properties will automatically be inherited? This could make the config files smaller with less for maintainers to specify. For example, status should never be inherited, but properties like name, description, docs should be (if the inheriting asset doesn't specify those properties thus increasing specificity.)

## Exclusion instead of inclusion

With the current approach, if we add a new schema key, all metadata files would have to be updated to inherit the new key. Instead, if it was more of an exclusion model where we assume all inheritable properties will be inherited by default, that would be more future proof.

### PR DESCRIPTION
**Pull request by mattrosno**
Closes #126 

Simplifies asset inheritance so it only has to be specified once at the library level and not in every asset. Also, it specifies inheritable defaults so if we add an asset key in the future, we could have that be auto inherited so we don't have to update all the asset yml files.

We don't have an issue for this, but per @jeffchew's recommendation when reviewing the schema, updated `private` to `noIndex` to more accurately reflect the property. If it's "not indexed", it doesn't show up in the global catalogs. Like how a web page doesn't show up in Google search results if the `noindex` meta tag/value is applied.

#### Changelog

**New**

- `inherits` library string key
- "Inheritable" column to the asset key for defaults
- `getLibraryVersionAsset` schema utility to parse identifier strings

**Changed**

- Renamed `private` to `noIndex` in the schema for libraries and assets
- Inheritance logic simplified as it references inherits at the library level and default inheritable properties
- Updated prototyping data

**Removed**

- `inherits` assert object key
- plaiceholder and sharp dependencies, and the thumbnail fetching, persisting, and rendering logic in `github.js` to clean up that file because we aren't using any of that functionality, because we'll need to do proper thumbnail support in the images epic in a later release

#### Testing / reviewing

You can view the updated docs in this commit: https://github.com/carbon-design-system/carbon-platform/blob/17f18ae266fbb5ae14a9f7b713da2a9b9e132d33/docs/resource-schemas.md

http://localhost:3000/assets/ibmdotcom-react

and

http://localhost:3000/assets/ibmdotcom-react/latest/back-to-top

Has inherited data.

### CODE REVIEW

**Review by francinelucca — CHANGES_REQUESTED** [review]
Nice code cleanup!! Just one comment on typeDefs and observation:
- Component cards don't seem to be punching out to external urls anymore, it's going to the detail page instead, intentional?

File: packages/schemas/carbon-resources.schema.json [hunk 1]
```diff
@@ -190,6 +151,10 @@
           "type": "string",
           "description": "Library description ideally between 50-160 characters in length. Use sentence-case capitalization. Defaults to the package.json description if not set here."
         },
+        "inherits ": {
```
Comment (francinelucca) [inline packages/schemas/carbon-resources.schema.json:154]:
```suggestion
        "inherits": {
```

File: services/web-app/types.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ export const libraryContentPropTypes = PropTypes.shape({
   license: PropTypes.oneOf(['', 'Apache-2.0', 'MIT']),
   name: PropTypes.string.isRequired,
   package: PropTypes.string,
-  private: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (mattrosno) [reply services/web-app/types.js:27]: Done!

**Comment by mattrosno** [comment]
> * Component cards don't seem to be punching out to external urls anymore, it's going to the detail page instead, intentional?

@francinelucca intentional - my original indexing of Carbon for IBM.com inherited `externalDocsUrl` everywhere. See https://github.ibm.com/matt-rosno/carbon-for-ibm-dotcom/search?q=externalDocsUrl&unscoped_q=externalDocsUrl

After thinking through this, it doesn't make much sense for that to be inheritable by default. I don't expect external docs to be the same for all sibling-framework implementations by default.

Maybe during our initial indexing that will result in some duplicate `externalDocsUrl` values, but that'll just be temporarily until we can remove those across all libraries/assets when we're able to do final migration and sunset legacy sites.

**Review by francinelucca — APPROVED** [review]

**Pull request merged**